src/services/codefixes/inferFromUsage.ts

```diff
@@ -275,11 +303,43 @@ namespace ts.codefix {
                 const typeExpression = createJSDocTypeExpression(typeNode);
                 const typeTag = isGetAccessorDeclaration(declaration) ? createJSDocReturnTag(typeExpression, "") : createJSDocTypeTag(typeExpression, "");
                 addJSDocTags(changes, sourceFile, parent, [typeTag]);
+                return;
             }
-            else {
-                changes.tryInsertTypeAnnotation(sourceFile, declaration, typeNode);
+            if (tryReplaceImportTypeNodeWithAutoImport(typeNode, changes, sourceFile, declaration, type, program, host, formatContext, preferences)) {

```

Comment:
```suggestion
            else if (!tryReplaceImportTypeNodeWithAutoImport(typeNode, changes, sourceFile, declaration, type, program, host, formatContext, preferences)) {
                changes.tryInsertTypeAnnotation(sourceFile, declaration, typeNode);
            }
```

(and drop the new `returns`)
I like this form a little better but neither are that great honestly.